pkgs/servers/mautrix-signal/default.nix

```
@@ -32,16 +35,30 @@ buildGoModule rec {
     })
   ];
 
-  buildInputs = (lib.optional (!withGoolm) olm) ++ [
-    # must match the version used in https://github.com/mautrix/signal/tree/main/pkg/libsignalgo
-    # see https://github.com/mautrix/signal/issues/401
-    libsignal-ffi
-  ];
+  buildInputs =
+    (lib.optional (!withGoolm) olm)
+    ++ (lib.optional withGoolm stdenv.cc.cc.lib)
+    ++ [
+      # must match the version used in https://github.com/mautrix/signal/tree/main/pkg/libsignalgo
+      # see https://github.com/mautrix/signal/issues/401
+      libsignal-ffi
+    ];
+
   tags = lib.optional withGoolm "goolm";
 
+  CGO_LDFLAGS = [ "-lstdc++" ];
+
   vendorHash = "sha256-bKQKO5RqgMrWq7NyNF1rj2CLp5SeBP80HWxF8MWnZ1U=";
 
-  doCheck = false;
+  doCheck = true;
+  preCheck = ''
+    # Needed by the tests to be able to find libstdc++
+    export LD_LIBRARY_PATH="${stdenv.cc.cc.lib}/lib:$LD_LIBRARY_PATH"
+  '';
+
+  passthru.tests.version = testers.testVersion {
+    package = mautrix-signal;
+  };
```

```suggestion
  nativeInstallCheckInputs = [ versionCheckHook ];
  versionCheckProgram = "${placeholder "out"}/bin/mautrix-signal";
  versionCheckProgramArg = [ "--version" ];
  doInstallCheck = true;
```

https://github.com/NixOS/nixpkgs/blob/master/doc/hooks/versionCheckHook.section.md (added a few months ago, harder to miss when building the package than the passthru test as it's part of the installPhase)

---

`versionCheckProgram` can actually be omitted as it defaults to `${placeholder "out"}/bin/${pname}`

---

Oh, that is nice! I'll change it, thanks!